src/auth/auth.cpp

```diff
@@ -560,13 +681,6 @@ std::vector<auth::User> Auth::AllUsersForRole(const std::string &rolename_orig)
 #ifdef MG_ENTERPRISE
 Auth::Result Auth::GrantDatabase(const std::string &db, const std::string &name, system::Transaction *system_tx) {
   using enum Auth::Result;
-  if (module_.IsUsed()) {
-    if (auto role = GetRole(name)) {
-      GrantDatabase(db, *role, system_tx);
-      return SUCCESS;
-    }
-    return NO_ROLE;
-  }
```

Comment:
How will this work?
The modules still return a role, so we have to look for a role, right?

Reply:
Yes, that’s what the function still does (see below):
```
  if (auto role = GetRole(name)) {
    GrantDatabase(db, *role, system_tx);
    return SUCCESS;
  }
```